### PR TITLE
HOP-2298 : Converted Token Replacement plugin from Chris Deptula

### PR DESCRIPTION
**Pull request by sramazzina**
HOP-2298 : Just finished convertion of Chris Deptula

### CODE REVIEW

File: plugins/transforms/tokenreplacement/src/main/doc/tokenreplacement.adoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+:documentationPath: /plugins/transforms/
+:language: en_US
+:page-alternativeEditUrl: https://github.com/apache/incubator-hop/edit/master/plugins/transforms/abort/src/main/doc/abort.adoc
```
Comment (hansva) [inline plugins/transforms/tokenreplacement/src/main/doc/tokenreplacement.adoc:3]: change the url to match the filename of this file

File: plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenReplacement.java [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+import java.io.Writer;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.util.Iterator;
+
+/**
+ * @author Chris
+ * @since 1-nov-2014
+ */
```
Comment (hansva) [inline plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenReplacement.java:53]: remove author

File: plugins/transforms/tokenreplacement/src/main/doc/tokenreplacement.adoc [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+== Options
+
+[width="90%", options="header"]
+|===
+|Option|Description
+|Transform name|Name of the transform.
+|Abort threshold|The threshold of number of rows after which to abort the pipeline. E.g. If threshold is 0, the abort transform will abort after seeing the first row. If threshold is 5, the abort transform will abort after seeing the sixth row.
+|Abort message|The message to put in the log upon aborting. If not filled in a default message will be used.
+|Always log|Always log the rows processed by the Abort transform. This allows the rows to be logged although the log level of the pipeline would normally not do it. This way you can always see in the log which rows caused the pipeline to abort.
+|===
```
Comment (hansva) [inline plugins/transforms/tokenreplacement/src/main/doc/tokenreplacement.adoc:19]: Just change the title and clean out the file and create a ticket if you are not writing the docs, then we have a placeholder and an action

File: plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenReplacementField.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+import org.apache.hop.core.Const;
+import org.apache.hop.core.injection.Injection;
+
+/**
+ * Describes a single field in a text file
+ *
+ * @author Chris
+ * @since 06-10-2014
+ *
```
Comment (hansva) [inline plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenReplacementField.java:29]: remove author

File: plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenResolver.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+package org.apache.hop.pipeline.transforms.tokenreplacement;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by openbi on 10/31/2014.
+ */
```
Comment (hansva) [inline plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenResolver.java:25]: remove author

**Review by hansva — CHANGES_REQUESTED** [review]
I'll do a functional test tomorrow, looks good

File: plugins/transforms/tokenreplacement/src/main/doc/tokenreplacement.adoc (hunk 1 above)
Comment (sramazzina) [reply plugins/transforms/tokenreplacement/src/main/doc/tokenreplacement.adoc:3]: I forget to cleanup this file. I agree to do this.

File: plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenReplacement.java (hunk 2 above)
Comment (sramazzina) [reply plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenReplacement.java:53]: I was thinking about this but I was unsure. I agree to remove.

File: plugins/transforms/tokenreplacement/src/main/doc/tokenreplacement.adoc (hunk 3 above)
Comment (sramazzina) [reply plugins/transforms/tokenreplacement/src/main/doc/tokenreplacement.adoc:19]: FIle has been cleaned up and ticket has been opened.

File: plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenReplacementField.java (hunk 4 above)
Comment (sramazzina) [reply plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenReplacementField.java:29]: Done

File: plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenResolver.java (hunk 5 above)
Comment (sramazzina) [reply plugins/transforms/tokenreplacement/src/main/java/org/apache/hop/pipeline/transforms/tokenreplacement/TokenResolver.java:25]: Done

**Pull request merged**